src/lumigo_tracer/lumigo_utils.py

```diff
@@ -324,9 +336,32 @@ def _publish_spans_to_kinesis(to_send: bytes, region: str) -> int:
             aws_access_key_id=Configuration.edge_kinesis_aws_access_key_id,
             aws_secret_access_key=Configuration.edge_kinesis_aws_secret_access_key,
         )
+    except Exception as err:
+        get_logger().exception("Failed to send spans to Kinesis", exc_info=err)
+        warn_client(f"Failed to send spans to Kinesis: {err}")
     return int((time.time() - start_time) * 1000)
 
 
+def _get_edge_kinesis_boto_client(region: str, aws_access_key_id: str, aws_secret_access_key: str):
+    global edge_kinesis_boto_client
+    if (
+        not edge_kinesis_boto_client
+        or os.environ.get("LUMIGO_KINEISIS_INITIALIZATION_CONNECTION", "").lower()  # noqa
```

Comment:
can you extract it to a util function?

Reply:
why do you have this key at all?
you're not initializing during cold start, so this flag is irrelevant.
In your implementation, this flag is actually "should_reuse_connection", which is different :)

Reply:
Ok, I'll rename the env variable